crates/goose/src/token_counter.rs

```diff
@@ -0,0 +1,103 @@
+use tokenizers::tokenizer::Tokenizer;
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+pub struct TokenCounter {
+    tokenizers: HashMap<String, Tokenizer>,
+}
+
+
+impl TokenCounter {
+    // static method to get the tokenizer files directory
+     fn tokenizer_files_dir() -> PathBuf {
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        path.push("../../tokenizer_files");
+        path
```

Comment:
not super sure where we'd want the tokenizer.json files to live

Reply:
are these not on the hub? or are we trying to avoid there being a download (not sure where if at all the hub downloads are cached on filesystem if they handle that)

https://docs.rs/tokenizers/latest/tokenizers/#loading-a-pretrained-tokenizer-from-the-hub

Reply:
`from_pretrained` requires adding hf-hub as a dependency & wanted to avoid that. https://github.com/huggingface/tokenizers/issues/1633